app/src/main/java/org/simple/clinic/patient/BusinessId.kt

```diff
@@ -35,39 +38,70 @@ data class BusinessId(
     val deletedAt: Instant?
 ) {
 
-  enum class IdentifierType {
-    BP_PASSPORT,
-    UNKNOWN;
+  sealed class IdentifierType {
+
+    object BpPassport : IdentifierType()
+
+    data class Unknown(val actual: String) : IdentifierType()
+
+    object TypeAdapter: SafeEnumTypeAdapter<IdentifierType>(
+        knownMappings = mapOf(
+            BpPassport to "simple_bp_passport"
+        ),
+        unknownStringToEnumConverter = { Unknown(it) },
+        unknownEnumToStringConverter = { (it as Unknown).actual }
+    )
 
     class RoomTypeConverter {
 
       @TypeConverter
-      fun fromStringValue(stringValue: String) = try {
-        IdentifierType.valueOf(stringValue)
-      } catch (e: IllegalArgumentException) {
-        UNKNOWN
-      }
+      fun toEnum(enum: String?) = TypeAdapter.toEnum(enum)
 
       @TypeConverter
-      fun toStringValue(identifierType: IdentifierType) = identifierType.name
+      fun fromEnum(enum: IdentifierType?) = TypeAdapter.fromEnum(enum)
+    }
+
+    class MoshiTypeAdapter {
+
+      @FromJson
+      fun toEnum(enum: String?) = TypeAdapter.toEnum(enum)
```

Comment:
Same comment as above. Rename `enum` -> `stringValue`

Reply:
Renamed to `value`.